Springboot basic app

koics97 opened the pull request:
A basic Springboot app

koics97 commented:
I don't know why is there an old project in this pull request.....

lyancsie commented on ClubMembersSortedCF/src/members/v2/Residence.java:
```diff
@@ -0,0 +1,65 @@
+    public void setAddress(String address) {
+        this.address = address;
+    }
+}
```
please pay attention and don't put redundant stuff in your pr

lyancsie commented on demo/src/main/java/com/example/demo/service/IndexService.java:
```diff
@@ -0,0 +1,14 @@
+@AllArgsConstructor
+public class IndexService{
+    public LocalDate getDate(){
+        return LocalDate.now();
```
love your logic

lyancsie commented on demo/src/main/java/com/example/demo/service/IndexService.java:
```diff
@@ -0,0 +1,14 @@
+
+@Service
+@AllArgsConstructor
+public class IndexService{
```
please make sure to autoformat your code

lyancsie commented on demo/src/main/java/com/example/demo/controller/PeopleController.java:
```diff
@@ -0,0 +1,35 @@
+public class PeopleController {
+
+    @Autowired
+    private PeopleRepository peopleRepository;
```
please don't directly autowire repo in the controller

lyancsie replied:
this could be `final` and you could use `@RequiredArgsConstructor` to autowire (field injection is not recommended)

lyancsie replied:
this comment applies everywhere where you autowire stuff

lyancsie commented on demo/src/main/java/com/example/demo/model/People.java:
```diff
@@ -0,0 +1,17 @@
+@Getter
+@Setter
+public class People {
+    @Id
```
If it is autogenerated, please make sure to add it here too

lyancsie commented on demo/src/main/java/com/example/demo/repository/PeopleRepository.java:
```diff
@@ -0,0 +1,13 @@
+import java.time.Period;
+import java.util.List;
+
+public interface PeopleRepository extends JpaRepository<People, Integer> {
```
integer or long?

lyancsie commented on demo/src/main/java/com/example/demo/model/People.java:
```diff
@@ -0,0 +1,17 @@
+@Setter
+public class People {
+    @Id
+    private int id;
```
integer or long?

lyancsie commented on demo/src/test/java/com/example/demo/SpringbootBasicApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+
+	@Test
+	void contextLoads() {
+	}
```
love your detailed test

lyancsie commented on demo/src/main/java/com/example/demo/controller/IndexController.java:
```diff
@@ -0,0 +1,20 @@
+public class IndexController {
+
+    @Autowired
+    private IndexService indexService;
```
https://github.com/koics97/Java-homework/pull/3/files#r1554557429

lyancsie commented on demo/src/main/java/com/example/demo/controller/PeopleController.java:
```diff
@@ -0,0 +1,35 @@
+    @GetMapping("/people3")
+    public List<People> getPeople(@RequestParam int age) {
+        return peopleRepository.findByAge(age);
+    }
```
an empty line would be welcome here

lyancsie commented on demo/src/main/java/com/example/demo/model/People.java:
```diff
@@ -0,0 +1,17 @@
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity(name = "something")
```
we could use a more descriptive name here

koics97 commented:
refactored code and made some changes:
-added PeopleService
-removed `@Autowire` from everywhere where it was used and replaced with `@RequiredArgsConstructor`

lyancsie commented:
@koics97 Fair and square job - please add unit tests for the service using the concepts we have learnt about `Mockito` and mocking. Otherwise, it looks really good
